### PR TITLE
feat(rad): add array sort operator

### PR DESCRIPTION
**Pull request by girazoki**
This PR adds sort functionality for array types (for integers and strings)

### CODE REVIEW

File: rad/src/operators/array.rs [hunk 1]
```diff
@@ -58,6 +62,44 @@ pub fn map(input: &RadonArray, args: &[Value]) -> Result<RadonTypes, RadError> {
...
+    let disc = input.inner_type();
+    let value = input.value();
+
+    let mut result = Vec::new();
+    // Distinguish each discriminant case
+    if disc == discriminant(&RadonTypes::from(RadonString::default())) {
+        result = value
+            .iter()
+            .cloned()
+            .sorted_by(|a, b| match (a, b) {
```
Comment (tmpolaczyk) [inline rad/src/operators/array.rs]: There is a `sorted_by_key` function which is more clear in this case, you just need to return the inner string.

```suggestion
            .sorted_by_key(|v| match v {
```

File: rad/src/operators/array.rs [hunk 2]
```diff
@@ -58,6 +62,44 @@ pub fn map(input: &RadonArray, args: &[Value]) -> Result<RadonTypes, RadError> {
...
+                (RadonTypes::Integer(a), RadonTypes::Integer(b)) => a.cmp(&b),
+                _ => unreachable!(),
+            })
+            .collect();
+    } else {
+        Err(RadError::UnsupportedSortOp {
+            inner_type: value[0].clone().radon_type_name(),
```
Comment (tmpolaczyk) [inline rad/src/operators/array.rs]: Maybe this will panic if `value` is empty? Add a test for sorting an empty array, it should return an empty array.

File: rad/src/error.rs [hunk 3]
```diff
@@ -71,6 +71,15 @@ pub enum RadError {
...
     UnsupportedReducer { inner_type: String, reducer: String },
+    /// The sort operator is not implemented for non-string arrays
+    #[fail(
+        display = "Sort is not supported for Array with inner type `{}`",
+        inner_type
+    )]
+    UnsupportedSortOp { inner_type: String },
+    /// The operator is not implemented for non-homogeneous arrays
+    #[fail(display = "Operator is not supported for Array with non homogeneous types")]
+    UnsupportedOpNonHomogeneous,
```
Comment (aesedepece) [inline rad/src/error.rs]: Let's take the operator as an string argument and print it here.

File: rad/src/operators/array.rs (hunk 1 above)
Comment (tmpolaczyk) [reply rad/src/operators/array.rs]: It turns out this is cannot be done because of lifetimes...

File: rad/src/error.rs [hunk 4]
```diff
@@ -77,6 +77,18 @@ pub enum RadError {
         reducer, inner_type
     )]
     UnsupportedReducer { inner_type: String, reducer: String },
+    /// The sort operator is not implemented for non-string arrays
+    #[fail(
+        display = "Sort is not supported for Array with inner type `{}`",
```
Comment (aesedepece) [inline rad/src/error.rs]:
```suggestion
        display = "ArraySort is not supported for RadonArray with inner type `{}`",
```

File: rad/src/error.rs [hunk 5]
```diff
@@ -77,6 +77,18 @@ pub enum RadError {
...
+        inner_type
+    )]
+    UnsupportedSortOp { inner_type: String },
+    /// The operator is not implemented for non-homogeneous arrays
+    #[fail(
+        display = "`{}` is not supported for Array with non homogeneous types",
```
Comment (aesedepece) [inline rad/src/error.rs]:
```suggestion
        display = "`{}` is not supported for RadonArray with non homogeneous types",
```

File: rad/src/operators/array.rs [hunk 6]
```diff
@@ -81,6 +85,50 @@ pub fn filter(input: &RadonArray, args: &[Value]) -> Result<RadonTypes, RadError
...
 
+pub fn sort(input: &RadonArray) -> Result<RadonArray, RadError> {
+    // if input is empty, return the array
+    if input.value().is_empty() {
+        return Ok(input.clone());
+    }
+    // Sort not applicable if not homogeneous
+    if !input.is_homogeneous() {
+        Err(RadError::UnsupportedOpNonHomogeneous {
+            operator: "Sort".to_string(),
```
Comment (aesedepece) [inline rad/src/operators/array.rs]:
```suggestion
            operator: "ArraySort".to_string(),
```

File: rad/src/operators/array.rs [hunk 7]
```diff
@@ -81,6 +81,53 @@ pub fn filter(input: &RadonArray, args: &[Value]) -> Result<RadonTypes, RadError
     Ok(RadonArray::from(result).into())
 }
 
+pub fn sort(input: &RadonArray, args: &[Value]) -> Result<RadonArray, RadError> {
+    let input_value = input.value();
+    let mapped_array = map(input, args)?;
+    let mapped_array = match mapped_array {
+        RadonTypes::Array(x) => x,
+        _ => Err(RadError::UnsupportedSortOp {
```
Comment (tmpolaczyk) [inline rad/src/operators/array.rs]: This is unreachable as long as map(array) returns an array.

File: rad/src/operators/array.rs [hunk 8]
```diff
@@ -229,3 +276,326 @@ fn test_filter_negative() {
...
+    let output = sort(&input, &script).unwrap_err();
+
+    assert_eq!(
+        output.to_string(),
+        "ArraySort is not supported for RadonArray with inner type `RadonFloat`"
+    )
+}
+
+#[test]
+fn test_operate_sort_string_2_arrays() {
```
Comment (tmpolaczyk) [inline rad/src/operators/array.rs]:
```suggestion
fn test_sort_string_2_arrays() {
```

Comment (tmpolaczyk) [reply rad/src/operators/array.rs]: I think `operate` is used for tests which directly use the `operate()` function, can you remove it from all this test names?
